draft-wang-ppm-differential-privacy.md

```diff
@@ -414,23 +459,204 @@ random seed into a sample from a given distribution.
 
 > TODO: Specify a Gaussian sampler
 
-## Randomized Response
+## Randomized Response {#rr}
 
 > TODO: Specify any primitives required for randomized response mechanisms
 
 # DP Mechanisms for VDAFs {#mechanisms}
 
 The section defines a generic interface for DP mechanisms VDAFs.
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
-
-## Client-DP for Prio3Histogram
-
-> TODO: Concretely describe a Client-DP mechanism for Prio3Histogram.
-
-## Aggregator-DP for Prio3Histogram
-
-> TODO: Concretely describe an Aggregator-DP mechanism for Prio3Histogram.
+We will define an interface `DpMechanism` that composes the following:
+
+* An optional Client-DP mechanism that adds noise to Clients' measurements.
+
+* An optional Aggregator-DP mechanism that adds noise to an Aggregator's
+  aggregate share, based on the number of measurements, and the minimum
+  batch size.
+
+* An optional debiasing step that removes the bias in DP primitives (i.e.
+  `DpPrimitive.debias`).
+
+The composition of Client- and Aggregator-DP mechanisms enforces the DP
+guarantee of a data collection task.
+
+~~~
+class DpMechanism:
+    Measurement
+    AggregateShare
+    AggregateResult
+    DebiasedAggregateResult
+
+    def maybe_add_noise_to_measurement(self,
+                                       meas: Measurement,
+                                       ) -> Measurement:
+        """
+        Add noise to measurement, if required by the Client-DP
+        mechanism.
+        """
+        pass
+
+    def maybe_add_noise_to_agg_share(self,
+                                     agg_share: AggregateShare,
+                                     meas_count: Unsigned,
+                                     min_batch_size: Unsigned,
```

Comment:
@cjpatton I changed it back to `min_batch_size`, because I think `batch_size` might mean the actual batch size.